amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -458,6 +466,90 @@ void runProcessLoop() {
         slog.resetInfoLogging();
     }
 
+    private boolean allParentShardLeasesDeleted(Lease lease, ShardInfo shardInfo) throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+        boolean allParentShardLeasesDeleted = true;
+        for (String parentShard : lease.parentShardIds()) {
+            if (leaseCoordinator.leaseRefresher().getLease(ShardInfo.getLeaseKey(shardInfo, parentShard)) != null) {
+                allParentShardLeasesDeleted = false;
+            }
+        }
+        return allParentShardLeasesDeleted;
+    }
+
+    private void cleanupCompletedShards(Collection<ShardInfo> completedShards)
+            throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+        
+        for (ShardInfo completedShard : completedShards) {
+            final Lease lease = leaseCoordinator.getCurrentlyHeldLease(ShardInfo.leaseKeyProvider.apply(completedShard));
```

Comment:
What happens if leases is not currently held by this lease owner does it return null ?

Reply:
Missed this case, in that case we need to just ignore that lease and wait for another worker to pick it up.